akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -23,39 +23,45 @@ import scala.util.Random
 
 object PhasedFusingActorMaterializer {
 
-  val Debug = false
+  val Debug = true // FIXME
 
   val DefaultPhase: Phase[Any] = new Phase[Any] {
-    override def apply(settings: ActorMaterializerSettings, materializer: PhasedFusingActorMaterializer): PhaseIsland[Any] =
-      new GraphStageIsland(settings, materializer).asInstanceOf[PhaseIsland[Any]]
+    override def apply(settings: ActorMaterializerSettings, materializer: PhasedFusingActorMaterializer,
+                       stageName: String): PhaseIsland[Any] =
+      new GraphStageIsland(settings, materializer, stageName).asInstanceOf[PhaseIsland[Any]]
   }
 
   val DefaultPhases: Map[IslandTag, Phase[Any]] = Map[IslandTag, Phase[Any]](
     SinkModuleIslandTag → new Phase[Any] {
-      override def apply(settings: ActorMaterializerSettings, materializer: PhasedFusingActorMaterializer): PhaseIsland[Any] =
-        (new SinkModulePhase(materializer)).asInstanceOf[PhaseIsland[Any]]
+      override def apply(settings: ActorMaterializerSettings, materializer: PhasedFusingActorMaterializer,
+                         stageName: String): PhaseIsland[Any] =
+        (new SinkModulePhase(materializer, stageName)).asInstanceOf[PhaseIsland[Any]]
     },
     SourceModuleIslandTag → new Phase[Any] {
-      override def apply(settings: ActorMaterializerSettings, materializer: PhasedFusingActorMaterializer): PhaseIsland[Any] =
-        new SourceModulePhase(materializer).asInstanceOf[PhaseIsland[Any]]
+      override def apply(settings: ActorMaterializerSettings, materializer: PhasedFusingActorMaterializer,
+                         stageName: String): PhaseIsland[Any] =
```

Comment:
It still feels wrong to me to pass in a string. Also, it is not `stageName` as an island can host many stages.